lib/authorizer/index.js

```diff
@@ -120,3 +120,96 @@ module.exports = {
     AuthLoader: AuthLoader,
     authorizeRequest: authorizeRequest
 };
+
+// Interface
+/**
+ * Interface for implementing auth handlers
+ *
+ * @interface AuthHandlerInterface
+ */
+
+// Interface functions
+/**
+ * This hook decides whether all the required paramaters are present in the auth or not.
+ * 3 things can happen depending on how it calls the [done]{@link authPreHookCallback} callback:
+ *
+ * 1. Calls with no error: Sign the request and proceed to send the request
+ * 2. Calls with error but no request: don't sign the request, but proceed to send the request
+ * 3. Calls with error and a request (for the possible formats see {@link authPreHookCallback}):
+ *      - send the intermediate request
+ *      - invoke the auth's [init]{@link AuthHandlerInterface#init} hook with the response of the intermediate request
+ *      - invoke the auth's [pre]{@link AuthHandlerInterface#pre} hook
```

Comment:
should this info (line 134-141) be copied in the doc for the hook also on [line 191](https://github.com/postmanlabs/postman-runtime/pull/372/files#diff-2390d9c3409fdd1cfa9b0fd818780c94R191)

Reply:
This explains the purpose and all the flows in the hook. We should also explain what the parameters mean and how to configure them in the callback docs.